xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/PropertyClass.java

```diff
@@ -195,6 +196,33 @@ public BaseProperty fromString(String value)
         return null;
     }
 
+    /**
+     * Retrieve the already existing base property based on current name, or create a new one.
+     *
+     * @return the existing {@link BaseProperty} or a new one.
+     * @since 17.3.0RC1
+     */
+    @Unstable
+    protected BaseProperty getCurrentOrNewProperty()
+    {
+        PropertyInterface propertyInterface = null;
+        if (getObject() != null && getName() != null) {
+            try {
+                propertyInterface = getObject().get(getName());
+            } catch (XWikiException e) {
+                // in theory it should never really happen, I'm not sure why the exception is thrown...
+                LOGGER.error(String.format("Error while retrieving property [%s]", getName()), e);
+            }
+        }
+        BaseProperty property;
+        if (propertyInterface instanceof BaseProperty propertyField) {
```

Comment:
I don't understand this code. `getObject` is a pretty misleading name, something like `getParent` would have been more accurate, and in the case of a Property**Class** the parent is the BaseClass. So ion theory `getObject().get(getName()) == this`.

Reply:
> So in theory getObject().get(getName()) == this

So you mean I shouldn't check the instanceof or I should at all call `getObject().get(getName())`? I'm doing the instanceof call to be extrasafe and also perform a null check. For the `getObject().get(getName())` call I actually need it to check if the instance is already registered or not as a property of the object, to see if I'm modifying an existing property or not.

Reply:
What I mean is that getObject does not return the object at all, it return the class (the parent of this PropertyClass).

Reply:
ok, so what should I call to detect if the property exists already?

Reply:
You just cannot at this level, if you want to check if the property exists already you should do it on BaseObject side.

Reply:
hmmm so I'll need another fromString API I guess